src/root/views/Country/Preparedness/EapApplication/index.tsx

```diff
@@ -0,0 +1,246 @@
+import {
+  listToMap,
+} from '@togglecorp/fujs';
+import useAlert from '#hooks/useAlert';
+import LanguageContext from '#root/languageContext';
+import { EapsFields } from './common';
+import Button from '#components/Button';
+import {
+  EapsApiFields,
+  overviewFields,
+  eventDetailsFields,
+  contactFields,
+} from './common';
+import {
+  useLazyRequest,
+} from '#utils/restRequest';
+import type { match as Match } from 'react-router-dom';
+
+import Page from '#components/Page';
+import Tab from '#components/Tabs/Tab';
+import TabList from '#components/Tabs/TabList';
+import TabPanel from '#components/Tabs/TabPanel';
+import Tabs from '#components/Tabs';
+import EapOverview from './EapOverview';
+import Contacts from './Contacts';
+import EarlyAction from './EarlyAction';
+import {
+  useForm,
+  getErrorObject,
+  PartialForm,
+} from '@togglecorp/toggle-form';
+import useEapFormOptions, { schema } from './useEapFormOptions';
+import styles from './styles.module.scss';
+
+const defaultFormValues: PartialForm<EapsFields> = {
+};
+
+interface EapsResponseFields {
+  id: number;
+}
+
+interface Props {
+  className?: string;
+  match: Match<{ eapId?: string }>;
+  history: History;
+  location: Location;
+}
+
+type StepTypes = 'eapOverview' | 'earlyActions' | 'contacts';
+const stepTypesToFieldsMap: {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  [key in StepTypes]: (keyof EapsFields)[];
+} = {
+  eapOverview: overviewFields,
+  earlyActions: eventDetailsFields,
+  contacts: contactFields,
+};
+
+function EapApplication(props: Props) {
+  const {
+    className,
+    match,
+    history,
+  } = props;
+
+  const alert = useAlert();
+  const { eapId } = match.params;
+  const { strings } = React.useContext(LanguageContext);
+
+  const {
+    value,
+    error,
+    setFieldValue: onValueChange,
+    validate,
+    setError: onErrorSet,
+    setValue: onValueSet,
+  } = useForm(schema, { value: defaultFormValues });
+
+  const {
+    fetchingCountries,
+    countryOptions,
+  } = useEapFormOptions(value);
+
+  const [currentStep, setCurrentStep] = React.useState<StepTypes>('eapOverview');
+  const submitButtonLabel = currentStep === 'contacts' ? strings.drefFormSaveButtonLabel : strings.drefFormContinueButtonLabel;
+  const shouldDisabledBackButton = currentStep === 'eapOverview';
+
+  const handleTabChange = React.useCallback((newStep: StepTypes) => {
+    setCurrentStep(newStep);
+  }, []);
+
+  const erroredTabs = React.useMemo(() => {
+    const tabs: {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      [key in StepTypes]: boolean;
+    } = {
+      eapOverview: false,
+      earlyActions: false,
+      contacts: false,
+    };
+
+    const tabKeys = (Object.keys(tabs)) as StepTypes[];
+    tabKeys.forEach((tabKey) => {
+      const currentFields = stepTypesToFieldsMap[tabKey];
+      const currentFieldsMap = listToMap(currentFields, d => d, d => true);
```

Comment:
Use appropriate naming instead of `d` throughout this function.
```suggestion
      const currentFieldsMap = listToMap(currentFields, field => field, field => true);
```